tests/utils.py

```diff
@@ -274,11 +274,48 @@ def clean_scsi_debug(scsi_debug_dev):
     except:
         pass
 
-def find_nvme_ctrl_devs_for_subnqn(subnqn):
+def _wait_for_nvme_controllers_ready(subnqn, timeout=2):
+    """
+    Wait for NVMe controllers with matching subsystem NQN to be in live state
+    
+    :param str subnqn: subsystem nqn to match controllers against
+    :param int timeout: timeout in seconds (default: 2)
+    :returns: True if ready, False if timeout
+    """
+    start_time = time.time()
+    wait_time = 0.1  # Start with 100ms
+    max_wait = 0.5   # Cap at 500ms
+    
+    while time.time() - start_time < timeout:
+        try:
+            for ctrl_path in glob.glob("/sys/class/nvme/nvme*/"):
+                state_file = os.path.join(ctrl_path, "state")
+                subsysnqn_file = os.path.join(ctrl_path, "subsysnqn")
+                try:
+                    state = read_file(state_file).strip()
+                    controller_subnqn = read_file(subsysnqn_file).strip()
+                    if state == "live" and controller_subnqn == subnqn:
+                        # Found a matching live controller, return immediately
+                        os.system("udevadm settle")
+                        return True
+                except:
+                    continue
+                
+        except:
+            pass
+        
+        time.sleep(wait_time)
+        wait_time = min(wait_time * 1.2, max_wait)  # Slower exponential backoff
```

Comment:
My [original comment](https://github.com/storaged-project/libblockdev/pull/1124#discussion_r2318148827) about this being unnecessarily complicated  is still valid here.